apps/library/include/gpc_memory_layout.h

```diff
@@ -0,0 +1,136 @@
+#ifndef _GPC_MEMORY_LAYOUT_H_
+#define _GPC_MEMORY_LAYOUT_H_
+
+/**
+ * Default values for memory layout constants. Actual values can be overridden by build configuration
+ * Used to generate LD script and may be included by entry point, so it must contain preprocessor
+ * directives only
+ */
+
+#include "gpc_defs.h"
+
+/**
+ * Number of agent ID bits in GPC memory address
+ */
+#ifndef GPC_MEM_AGENT_ID_BITS
+    #define GPC_MEM_AGENT_ID_BITS (8)
+#endif
+
+// Base of CR space address (CR region at the right place, other bits zero)
+#define GPC_MEM_REGION_OFFSET (32 - 2 - GPC_MEM_AGENT_ID_BITS)
+#define GPC_MEM_REGION_BASE(region) ((region) << GPC_MEM_REGION_OFFSET)
+
+/**
+ * I_MEM size
+ */
+#ifndef GPC_MEM_I_MEM_SIZE
+#define GPC_MEM_I_MEM_SIZE 0x800
+#endif
+
+/**
+ * D_MEM size
+ */
+#ifndef GPC_MEM_D_MEM_SIZE
+#define GPC_MEM_D_MEM_SIZE 0x800
```

Comment:
Currently, in design  "HW" the default value is 4KB. (0x1000)

Reply:
I can set the default value to whatever fits our needs best. Anyway it can be overridden in build time by adding a parameter to the build command